packages/financial-templates-lib/test/price-feed/DefiPulseTotalPriceFeed.js

```diff
@@ -0,0 +1,87 @@
+require("dotenv");
+const { DefiPulseTotalPriceFeed } = require("../../src/price-feed/DefiPulseTotalPriceFeed");
+const { NetworkerMock } = require("../../src/price-feed/NetworkerMock");
+const winston = require("winston");
+var Web3 = require("web3");
+var web3 = new Web3("http://localhost:8545");
+
+contract("DefiPulseTotalPriceFeed.js", function() {
+  let defiPulseTotalPriceFeed;
+  let mockTime = 1611583300;
+  let networker;
+
+  const apiKey = 'test-api-key';
+
+  const lookback = 3600 * 24 * 7;
+
+  const getTime = () => mockTime;
+  const minTimeBetweenUpdates = 600; // every 10 minutes
+
+  // Fake data to inject.
+  const validResponses = [
+    [
+      { timestamp: "1611572400", tvlUSD: 24780000000 },
+      { timestamp: "1611576000", tvlUSD: 23500000000 },
+      { timestamp: "1611579600", tvlUSD: 22250000000 },
+      { timestamp: "1611583200", tvlUSD: 25100001000 }
+    ]
+  ];
+
+  beforeEach(async function() {
+    networker = new NetworkerMock();
+    const dummyLogger = winston.createLogger({
+      level: "info",
+      transports: [new winston.transports.Console()]
+    });
+
+    defiPulseTotalPriceFeed = new DefiPulseTotalPriceFeed(
+      dummyLogger,
+      web3,
+      apiKey,
+      lookback,
+      networker,
+      getTime,
+      minTimeBetweenUpdates
+    );
+  });
+
+  it("No update", async function() {
+    assert.equal(defiPulseTotalPriceFeed.getCurrentPrice(), undefined);
+    assert.isTrue(await defiPulseTotalPriceFeed.getHistoricalPrice(1000).catch(() => true));
+    assert.equal(defiPulseTotalPriceFeed.getLastUpdateTime(), undefined);
+  });
+
+  it("Basic historical price", async function() {
```

Comment:
Can you add a test for non-18 decimals just to verify that that scaling function you added works correctly?

Reply:
ye, please add a test for this :)